packages/block/src/header.ts

```diff
@@ -351,15 +351,10 @@ export class BlockHeader {
   }
 
   private async _getBlockByHash(blockchain: Blockchain, hash: Buffer): Promise<Block | undefined> {
-    return new Promise((resolve, reject) => {
-      blockchain.getBlock(hash, (err, block) => {
-        if (err) {
-          reject(err)
-          return
-        }
-
-        resolve(block)
-      })
-    })
```

Comment:
For the record: this is changing the API of the `block` library in the form that `validate()` - calling into `_getBlockByHash()` now only works with the yet to be released promisifed `Blockchain` version.

Wonder if this would be a good case for adding a `peerDependency` (see [Peer Dependencies | Node.js](https://nodejs.org/es/blog/npm/peer-dependencies/)) entry in the `Block` `package.json` referencing the `Blockchain` version? // cc @alcuadrado 

Do we want to do a more systematic approach and generally introduce `peerDependency` references within the monorepo scope? These inter-library compatibility insecurities are a reoccurring problem.

Reply:
If someone reading this think this is a good idea eventually directly open an issue on this, optimally (but leave if no time) directly with a TODO list with the listed directed references/dependencies.

Reply:
This should also be added as a linked reference to the v5 release TODO list (if we agree upon), since it would make very much sense to add this along the new releases.